libtenzir/include/tenzir/detail/base64.hpp

```diff
@@ -61,6 +61,7 @@ size_t encode(void* dst, const void* src, size_t len);
 /// @param str The string to encode.
 /// @returns The Base64-encoded version of *str*.
 std::string encode(std::string_view str);
+std::string encode(std::basic_string_view<std::byte> str);
```

Comment:
Shouldn't this be a `std::span<const std::byte>` for the argument? I don't think `std::char_traits<std::byte>` is well-defined.

Reply:
I was also not so sure about that `std::char_traits<std::byte>`, but we are currently using `std::basic_string_view<std::byte>` for `view<blob>`, and this is not implicitly convertible to `std::span<const std::byte>`. I can add a generic version that also works with `std::span<const std::byte>` if that's preferrable.

Reply:
And the reason for using `basic_string[_view]` over `vector` and `span` is that many other places in the code base assume that many types are addable, comparable, etc.

Reply:
I trust your judgement on this. I honestly don't know what's better.